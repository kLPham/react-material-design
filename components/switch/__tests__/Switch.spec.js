import React from 'react';
import { shallow } from 'enzyme';
import Switch from '../index';

jest.mock('uuid', () => ({
    v4: jest.fn(() => '2e732f05-f466-4fba-acb5-2f3be58daf75'),
}));
describe('<Switch />', () => {
    it('should render', () => {
        const wrapper = shallow(
            <Switch />,
    );
        expect(wrapper).toMatchSnapshot();
    });
    it('should render with a label', () => {
        const label = 'this is my label';
        const wrapper = shallow(
            <Switch label={label} />,
    );
        expect(wrapper).toMatchSnapshot();
    });
    it('should render with a label', () => {
        const label = 'this is my label';
        const disabled = true;
        const wrapper = shallow(
            <Switch label={label} disabled={disabled} />,
    );
        expect(wrapper).toMatchSnapshot();
    });
});
