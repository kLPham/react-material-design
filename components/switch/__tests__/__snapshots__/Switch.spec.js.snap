// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render 1`] = `
<div>
  <div
    className="mdc-switch"
  >
    <input
      className="mdc-switch__native-control"
      id="mdc-switch--2e732f05-f466-4fba-acb5-2f3be58daf75"
      type="checkbox"
    />
    <div
      className="mdc-switch__background"
    >
      <div
        className="mdc-switch__knob"
      />
    </div>
  </div>
</div>
`;

exports[`<Switch /> should render with a label 1`] = `
<div>
  <div
    className="mdc-switch"
  >
    <input
      className="mdc-switch__native-control"
      id="mdc-switch--2e732f05-f466-4fba-acb5-2f3be58daf75"
      type="checkbox"
    />
    <div
      className="mdc-switch__background"
    >
      <div
        className="mdc-switch__knob"
      />
    </div>
  </div>
  <label
    className="mdc-switch-label"
    htmlFor="mdc-switch--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
</div>
`;

exports[`<Switch /> should render with a label 2`] = `
<div>
  <div
    className="mdc-switch mdc-switch--disabled"
  >
    <input
      className="mdc-switch__native-control"
      disabled={true}
      id="mdc-switch--2e732f05-f466-4fba-acb5-2f3be58daf75"
      type="checkbox"
    />
    <div
      className="mdc-switch__background"
    >
      <div
        className="mdc-switch__knob"
      />
    </div>
  </div>
  <label
    className="mdc-switch-label"
    htmlFor="mdc-switch--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    this is my label
  </label>
</div>
`;
